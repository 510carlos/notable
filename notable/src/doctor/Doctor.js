import React, { useEffect, useState } from 'react';
import { getApointment } from './Doctors.helpers';
import { Apointments, Information } from './components';

const Doctor = ({match}) => {
    const id = match.params.id || -1;

    const [apt, setApt] = useState([]);

    useEffect( () => {
        let mounted = true;
        getApointment(id).then( result => {
        const { data } = result;
        if (mounted)
            setApt(data)
        });
        return () => {
            mounted = false;
        }
    }, [id]);

    return (
        <>
            <Information id={id} />
            <Apointments apt={apt} />
        </>
    )
}

export default Doctor;