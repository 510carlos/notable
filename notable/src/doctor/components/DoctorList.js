import React from 'react';
import { Redirect } from 'react-router';
import { Button, Menu, Icon } from 'antd';
import { useDoctorContext } from '../Doctor.context';
import { StoreDocID } from '../Doctors.hooks';

const { SubMenu } = Menu;

const DoctorList = () => {
    const [docId, setId] = StoreDocID();
    const doctorsList = useDoctorContext();

    const handleClick = e => {
        const id = e.key;
        if(id)
            setId(id)
    }

    if (docId !== -1)
        return <Redirect to={`/doctor/${docId}`} />

    const items = [];
    if(doctorsList && doctorsList.length !== 0)
        doctorsList.map((doc, id) =>
            items.push(<Menu.Item key={id}>{doc.name}</Menu.Item>)
        )

    return (
        <Menu
            mode="inline"
            defaultSelectedKeys={['1']}
            defaultOpenKeys={['sub1']}
            style={{ height: '100%', borderRight: 0 }}
            onClick={handleClick}
        >
            <SubMenu
            key="sub1"
            title={
                <span>
                <Icon type="user" />
                Physicians
                </span>
            }
            >
            {items}
            </SubMenu>
            <Button block={true} type="primary">Logout</Button>
        </Menu>
    )
};

export default DoctorList;