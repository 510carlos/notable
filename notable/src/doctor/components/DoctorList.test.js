import * as React from 'react';

import { shallow, mount } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import DoctorList from './DoctorList';

import { Menu } from 'antd';

import * as DoctorContext from '../Doctor.context';

const dom = <DoctorList />;

const contextValues = [{"id":1,"name":"Hibbert, Julius","email":"kriger@notablehealth.com"},{"id":2,"name":"Krieger, Algemap","email":"kriger@notablehealth.com"},{"id":3,"name":"Riviera, Nick","email":"kriger@notablehealth.com"}];
jest.spyOn(DoctorContext, 'useDoctorContext')
    .mockImplementation(() => contextValues);

describe('Unit tests for DoctorList', () => {
    it('Renders', () => {
        shallow(dom)
    })

    it('Renders with snapshot', () => {
        const wrapper = shallow(dom);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    })

    it('It renders with the correct props', () => {
        const wrapper = shallow(dom);
        let doctors = [],
            doctorsNames = [];

        doctors = wrapper.find(Menu.Item);

        let i = 0;
        doctors.forEach(node => {
            expect(contextValues[i].name).toEqual(node.props().children)
            i = i + 1;
        });
    })
})