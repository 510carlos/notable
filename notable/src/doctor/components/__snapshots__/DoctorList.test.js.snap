// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for DoctorList Renders with snapshot 1`] = `
<Menu
  defaultOpenKeys={
    Array [
      "sub1",
    ]
  }
  defaultSelectedKeys={
    Array [
      "1",
    ]
  }
  mode="inline"
  onClick={[Function]}
  style={
    Object {
      "borderRight": 0,
      "height": "100%",
    }
  }
>
  <SubMenu
    key="sub1"
    title={
      <span>
        <Icon
          type="user"
        />
        Physicians
      </span>
    }
  >
    <MenuItem
      key="0"
    >
      Hibbert, Julius
    </MenuItem>
    <MenuItem
      key="1"
    >
      Krieger, Algemap
    </MenuItem>
    <MenuItem
      key="2"
    >
      Riviera, Nick
    </MenuItem>
  </SubMenu>
  <Button
    block={true}
    ghost={false}
    htmlType="button"
    loading={false}
    type="primary"
  >
    Logout
  </Button>
</Menu>
`;
